text/078-tbc-title.md

```diff
@@ -0,0 +1,508 @@
+# RFC 07?: TBC - Adopting Stimulus
+
+- RFC: 7?
+- Author: LB (Ben) Johnston
+- Created: 2022-06-05
+- Last Modified: 2022-06-05
+
+- [RFC 07?: TBC - Adopting Stimulus](#rfc-07-tbc---adopting-stimulus)
+  - [Abstract](#abstract)
+    - [Overview](#overview)
+    - [Primary Goals](#primary-goals)
+    - [Secondary Goals](#secondary-goals)
+    - [History & Initial Research](#history--initial-research)
+  - [Specification](#specification)
+    - [Initial Implementation Overview](#initial-implementation-overview)
+    - [Implementation roadmap](#implementation-roadmap)
+    - [Proposed Documentation](#proposed-documentation)
+  - [Open Questions](#open-questions)
+    - [Storybook compatibility](#storybook-compatibility)
+    - [API for a controller definition as an object](#api-for-a-controller-definition-as-an-object)
+    - [Dispatched event names](#dispatched-event-names)
+    - [Prefix on controllers](#prefix-on-controllers)
+    - [TypeScript verbosity](#typescript-verbosity)
+    - [Handling animations](#handling-animations)
+  - [Answered questions](#answered-questions)
+    - [Why use a HTML first approach](#why-use-a-html-first-approach)
+    - [Why not more React](#why-not-more-react)
+    - [Why Not use Vue, Svelte, Angular, or Solid etc](#why-not-use-vue-svelte-angular-or-solid-etc)
+    - [Why Not Alpine.js](#why-not-alpinejs)
+    - [Why Stimulus and not something else](#why-stimulus-and-not-something-else)
+  - [Additional Information](#additional-information)
+    - [Links](#links)
+    - [Stimulus in the wild](#stimulus-in-the-wild)
+    - [Future possibilities](#future-possibilities)
+
+## Abstract
+
+This RFC proposes that Wagtail adopt [Stimulus](https://stimulus.hotwired.dev/) for its lightweight JavaScript framework.
+
+The aim of this lightweight framework is to provide a consistent, predictable and extensible way to build interactive JavasScript UI elements.
+
+This is so that that Wagtail has a solid alternative to jQuery widgets, inline script tag usage while allowing those that use Wagtail to develop UI customisations to be supported.
+
+> 🚧 A prototype implementation of this proposal is available for review and testing on https://github.com/lb-/wagtail-rfcs/tree/rfc/stimulus.
+
+### Overview
+
+We want to to see what may suitable for the middle ground of interactivity where React may not be an ideal option. We want to make it easy for those building Wagtail and building with Wagtail to think in HTML first (leveraging Django) but make a great experience for those editing width Wagtail.
```

Comment:
> We want to make it easy for those building Wagtail and building with Wagtail to think in HTML first (leveraging Django)

I think it's worth discussing this, maybe on the full RFC though (or in a UI team meeting).

Wagtail isn't entirely "HTML first" at the moment, many things are defined in Python and rendered by Wagtail with either a HTML template (tables, forms, panels) or converted to JavaScript objects with Telepath and rendered directly with JavaScript (sidebar and streamfield).

Reply:
True. Html first may be a bit of a bloated term - along with vanilla JS. 

What I am trying to convey is that we want to extend on Django (including html templates) instead of another layer that is separate from Django (where suitable). 

I cannot speak for the general direction of Wagtail long term though - I'm only basing this on the recent Wagtail space talk from Matt, my own experience of Wagtail and comments like this. 
https://github.com/wagtail/wagtail/discussions/8128#discussioncomment-2365615

If Wagtail is heading towards maybe a SPA direction - then this RFC is not probably worth pursuing. 

Hopefully this RFC will help clarify things either way.